docs/src/threadpool.rst

```diff
@@ -52,15 +96,59 @@ Public members
 
 .. seealso:: The :c:type:`uv_req_t` members also apply.
 
+.. c:member:: uv_executor_submit_func uv_executor_t.submit
+
+.. c:member:: uv_executor_cancel_func uv_executor_t.cancel
+
+.. c:member:: uv_work_type uv_work_options_t.type
+
+    Type of request. Readonly.
+
+    ::
+
+        typedef enum {
+            UV_WORK_UNKNOWN = 0,
+            UV_WORK_FS,
+            UV_WORK_DNS,
+            UV_WORK_USER_IO,
+            UV_WORK_USER_CPU,
+            UV_WORK_PRIVATE,
+            UV_WORK_TYPE_MAX
+        } uv_work_type;
+
+.. c:member:: int uv_work_options_t.priority
+
+    Suggested for use in user-defined executor.
+    Has no effect on libuv's default executor.
+
+.. c:member:: int uv_work_options_t.cancelable
+
+    Boolean.
+    If truth-y, it is safe to abort this work while it is being handled
+    by a thread.
```

Comment:
Suggestion: s/truth-y/nonzero/

It might be good to specify what `abort` means? As I understand it, currently libuv only cancels work requests that are not yet executing… does this map to that 1:1?

Reply:
> currently libuv only cancels work requests that are not yet executing

Yes. This has proved unfortunate in the case of DNS requests + shutdown of Node.js. See for example [here](https://github.com/libuv/libuv/issues/203#issuecomment-74080305) and [here](https://github.com/nodejs/node/issues/795#issuecomment-74063491).

My intention is to permit an executor to cancel work requests that have been assigned to threads, if they indicate that they are cancelable and if the executor so desires.